sdk/keyvault/keyvault-admin/test/internal/userAgent.spec.ts

```diff
@@ -16,7 +17,32 @@ describe("Key Vault Admin's user agent (only in Node, because of fs)", function
   });
 
   it("SDK_VERSION and packageVersion should match", async function () {
-    assert.equal(SDK_VERSION, packageVersion);
+    let userAgent: string | undefined;
+    const client = new KeyVaultAccessControlClient(
+      "https://myvault.vault.azure.net",
+      {} as TokenCredential,
+      {
+        additionalPolicies: [
+          {
+            policy: {
+              name: "fetchUserAgent",
+              sendRequest: (request, next) => {
+                userAgent = request.headers.get("user-agent");
+                return next(request);
+              },
+            },
+            position: "perCall",
+          },
+        ],
+      }
+    );
+    try {
+      await client.getRoleAssignment("/", "");
+    } catch {
+      // no-op, we don't care about the response, only the user-agent header
+    }
+    assert.exists(userAgent);
+    assert.include(userAgent!, `azsdk-js-keyvault-admin/${SDK_VERSION}`);
```

Comment:
`exists` checks are kinda hard to figure out from logs, so maybe

```suggestion
    if (!userAgent) {
       assert.fail("Expected a User-Agent header to be sent");
    } else {
       assert.include(userAgent, `azsdk-js-keyvault-admin/${SDK_VERSION}`, "User-Agent header doesn't include the correct SDK version");
    }
```

Reply:
I ended up going with `assert.exists(userAgent, "Expected a User-Agent header to be sent");` which will output the same message 👍 

Agreed that "expected undefined to exist" is not easy to debug 😄